extras/beast/unit_test/suite_info.hpp

```diff
@@ -52,61 +63,58 @@ class suite_info
         return library_;
     }
 
-    /** Returns `true` if this suite only runs manually. */
+    /// Returns `true` if this suite only runs manually.
     bool
     manual() const
     {
         return m_manual;
     }
 
-    /** Return the canonical suite name as a string. */
-    template <class = void>
+    /// Return the canonical suite name as a string.
     std::string
-    full_name() const;
+    full_name() const
+    {
+        return library_ + "." + module_ + "." + name_;
+    }
 
-    /** Run a new instance of the associated test suite. */
+    /// Run a new instance of the associated test suite.
     void
-    run (runner& r) const
+    run(runner& r) const
     {
         m_run (r);
     }
 };
 
 //------------------------------------------------------------------------------
 
-template <class>
-suite_info::suite_info (std::string const& name, std::string const& module,
-        std::string const& library, bool manual, run_type run)
-    : name_ (name)
-    , module_ (module)
-    , library_ (library)
-    , m_manual (manual)
-    , m_run (std::move (run))
-{
-}
-
-template <class>
-std::string
-suite_info::full_name() const
-{
-    return library_ + "." + module_ + "." + name_;
-}
-
 inline
 bool
-operator< (suite_info const& lhs, suite_info const& rhs)
+operator<(suite_info const& lhs, suite_info const& rhs)
 {
-    return lhs.full_name() < rhs.full_name();
+    auto const& ls = lhs.full_name();
+    auto const& rs = rhs.full_name();
+    return std::strcmp(ls.c_str(), rs.c_str()) < 0;
```

Comment:
`full_name` does string concatenation, which may alloc. I don't like that in a compare. We could use `std::tuple` to do the comparison without allocating:
```   
    friend
    bool
    operator<(suite_info const& lhs, suite_info const& rhs)
    {
        return std::tie (lhs.name_, lhs.module_, lhs.library_) <
            std::tie (rhs.name_, rhs.module_, rhs.library_);
    }
```